Nodejs/Product/Nodejs/TypingsAcquisition.cs

```diff
@@ -0,0 +1,123 @@
+            return await DownloadTypings(
+                pathToRootNpmDirectory,
+                pathToRootProjectDirectory,
+                newPackages,
+                redirector);
+        }
+
+        private static async Task<bool> DownloadTypings(
+            string pathToRootNpmDirectory,
+            string pathToRootProjectDirectory,
+            IEnumerable<IPackage> packages,
+            Redirector redirector) {
+            if (!packages.Any()) {
+                return true;
+            }
+
+            var tsdPath = GetTsdPath(pathToRootNpmDirectory);
+            if (tsdPath == null) {
+                if (redirector != null)
+                    redirector.WriteErrorLine(SR.GetString(SR.TsdNotInstalledError));
+                return false;
+            }
+
+            using (var process = ProcessOutput.Run(
+                tsdPath,
+                TsdInstallArguments(packages),
+                pathToRootProjectDirectory,
+                null,
+                false,
+                redirector,
+                quoteArgs: true)) {
+                if (!process.IsStarted) {
+                    // Process failed to start, and any exception message has
+                    // already been sent through the redirector
+                    if (redirector != null) {
+                        redirector.WriteErrorLine("could not start tsd");
+                    }
+                    return false;
+                }
+                var i = await process;
+                if (i == 0) {
+                    if (redirector != null) {
+                        redirector.WriteLine(SR.GetString(SR.TsdInstallCompleted));
+                    }
+                    return true;
+                } else {
+                    process.Kill();
+                    if (redirector != null) {
+                        redirector.WriteErrorLine(SR.GetString(SR.TsdInstallErrorOccurred));
+                    }
+                    return false;
+                }
+            }
+        }
+
+        private static string GetTsdPath(string pathToRootNpmDirectory) {
+            var path = Path.Combine(pathToRootNpmDirectory, TsdExe);
+            return File.Exists(path) ? path : null;
+        }
+
+        private static string GetPackageTsdName(IPackage package) {
+            return package.Name;
+        }
+
+        private static IEnumerable<string> TsdInstallArguments(IEnumerable<IPackage> packages) {
+            return new[] { "install", }.Concat(packages.Select(GetPackageTsdName)).Concat(new[] { "--save" });
+        }
+
+        private static IEnumerable<IPackage> TypingsToAcquire(string pathToRootProjectDirectory, IEnumerable<IPackage> packages) {
+            var currentTypings = new HashSet<string>(CurrentTypingsPackages(pathToRootProjectDirectory));
+            return packages.Where(package =>
+                !currentTypings.Contains(GetPackageTsdName(package)));
```

Comment:
Let's simplify to `package.Name` for now, and create `GetPackageTsdName` when it's necessary. Consider the experience of debugging through this code... the best case scenario is that you get to just hover on the variable to see the input, and unfortunately you can't do that with a method like this.